analyses/cell-type-dsrct/exploratory_analysis/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,440 @@
+  p <- ggplot(umap_df_list[[i]], aes(x = UMAP1, y = UMAP2, color = gene_expression)) +
+    geom_point(alpha = 0.8, size = 0.1) +
+    facet_wrap(vars(gene_symbol)) +
+    scale_color_viridis_c() +
+    labs(color = "Log-normalized gene expression") +
+    # remove axis numbers and background grid
+    scale_x_continuous(labels = NULL, breaks = NULL) +
+    scale_y_continuous(labels = NULL, breaks = NULL) +
+    theme(
+      aspect.ratio = 1,
+      legend.position = "bottom",
+      axis.title = element_text(size = 9, color = "black"),
+      strip.text = element_text(size = 8),
+      legend.title = element_text(size = 9),
+      legend.text = element_text(size = 8)
+    ) +
+    guides(colour = guide_colorbar(title.position = "bottom", title.hjust = 0.5)) +
+    ggtitle(names(umap_df_list)[i])
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      '../plots/marker_expression_',
+      names(umap_df_list)[i],
+      '.png'
+    ),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+}
+  
+
+```
+
+
+In my experience, ST6GALNAC5 is a strong marker for DSRCT in single-cell data. As can be seen, several of the samples have expression of ST6GALNAC5, but some do not, such as SCPCS000731 and SCPCS000729. In fact, these SCPCS000729 contains  low number of cells. These  samples in particular are the PDX samples and I will be excluding SCPCS000729 from further analyses. 
+
+```{r}
+umap_df_list_excluded <- umap_df_list[!(names(umap_df_list) %in% c('SCPCS000731'))]
+```
+
+
+We can also look at the distributions for each marker gene. 
+I would expect to see some sort of bimodal distribution separating cells that do and do not have expression of the marker gene. What is clear from these plots that ST6GALNAC5, CACNA2D2, PTPRQ, IQCJ-SCHIP1,  show a bi modal distribution among the cells. To some extent, we do see expression of the other markers. 
+
+```{r}
+for(i in 1:length(umap_df_list_excluded)) {
+  p <- ggplot(umap_df_list_excluded[[i]],
+              aes(x = gene_expression, fill = gene_symbol)) +
+    geom_density() +
+    facet_wrap(vars(gene_symbol), scales = 'free_y') +
+    theme(legend.position = "none") + 
+    ggtitle(names(umap_df_list_excluded)[i])
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      '../plots/marker_distribution_',
+      names(umap_df_list)[i],
+      '.png'
+    ),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+}
+```
+
+Although we see some slight semblance of a bimodal distribution for most marker genes, it is hard to see and we cannot directly compare gene expression values across genes. 
+This would make it hard to identify a cut off to categorize cells as expression or not expressing the marker gene. 
+
+Now we will transform each of the gene expression vectors by generating z-scores. 
+Then we might be able to find a cut off we can use across samples for if marker genes are present in a cell or not. 
```

Comment:
I think this is fine for a rough look, but it is quite brittle. As you note later, some samples may contain only tumor cells. In that case, using the zscore may not be a good measure; what we really want is to find the division between the bimodal peaks. This is beyond the scope of this PR, but for your final classifications I would not expect you to want to use z-scoring.